core-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java

```diff
@@ -250,15 +254,27 @@ else if (mySyncResult.needsSync()) {
       myWarningLabel2.setVisible(false);
       myInfoPanel.setVisible(true);
     }
-    else if (!mySyncResult.isDeterminable()) {
-      setOKButtonText(getIsContinued() ? GctBundle.getString("clouddebug.continueanyway") : GctBundle.getString("clouddebug.attach.anyway"));
+    else if (mySyncResult.needsSync() && mySyncResult.getTargetSyncSHA() == null) {
+      setOKButtonText(isContinued() ? GctBundle.getString("clouddebug.continueanyway") : GctBundle.getString("clouddebug.attach.anyway"));
+      myWarningLabel.setVisible(true);
+      myWarningLabel2.setVisible(true);
+      myInfoPanel.setVisible(true);
+      myWarningLabel2.setText("Could not locate a local Git repository with matching SHA");
+    }
+    else if (mySyncResult.getTargetRepository() == null && mySyncResult.getCloudRepository() == null) {
+      setOKButtonText(isContinued() ? GctBundle.getString("clouddebug.continueanyway") : GctBundle.getString("clouddebug.attach.anyway"));
       myWarningLabel.setVisible(true);
       myWarningLabel2.setVisible(true);
       myInfoPanel.setVisible(true);
-      myWarningLabel2.setText("Could not verify that current source matches module.");
+      if (mySyncResult.getRepositoryType() != null) {
+        myWarningLabel2.setText(mySyncResult.getRepositoryType() + " repositories are not supported.");
```

Comment:
why are you not supporting other git repos when the code is exactly the same?
You should just be getting the SHA from those source contexts as you do from the cloud repo.